cmake/FindFreeRTOS.cmake

```diff
@@ -1,122 +1,216 @@
+        add_library(${FreeRTOS_NAMESPACE}::StreamBuffer INTERFACE IMPORTED)
+        target_sources(${FreeRTOS_NAMESPACE}::StreamBuffer INTERFACE "${FreeRTOS_SOURCE_DIR}/stream_buffer.c")
+        target_link_libraries(${FreeRTOS_NAMESPACE}::StreamBuffer INTERFACE FreeRTOS)
+    endif()
+
+    if(NOT (TARGET ${FreeRTOS_NAMESPACE}::Timers))
+        add_library(${FreeRTOS_NAMESPACE}::Timers INTERFACE IMPORTED)
+        target_sources(${FreeRTOS_NAMESPACE}::Timers INTERFACE "${FreeRTOS_SOURCE_DIR}/timers.c")
+        target_link_libraries(${FreeRTOS_NAMESPACE}::Timers INTERFACE FreeRTOS)
+    endif()
+
+    foreach(HEAP ${FreeRTOS_HEAPS})
+        if(NOT (TARGET ${FreeRTOS_NAMESPACE}::Heap::${HEAP}))
+            add_library(${FreeRTOS_NAMESPACE}::Heap::${HEAP} INTERFACE IMPORTED)
+            target_sources(${FreeRTOS_NAMESPACE}::Heap::${HEAP} INTERFACE "${FreeRTOS_SOURCE_DIR}/portable/MemMang/heap_${HEAP}.c")
+            target_link_libraries(${FreeRTOS_NAMESPACE}::Heap::${HEAP} INTERFACE FreeRTOS)
+        endif()
+    endforeach()
+
+    foreach(PORT ${FreeRTOS_FIND_COMPONENTS_PORTS})
+        find_path(FreeRTOS_${PORT}_PATH
+            NAMES portmacro.h
+            PATHS "${FREERTOS_PATH}" "${FREERTOS_PATH}/FreeRTOS" 
+            PATH_SUFFIXES
+                "portable/GCC/${PORT}/r0p1"
+                "portable/GCC/${PORT}"
+                "Source/portable/GCC/${PORT}"
+                "Source/portable/GCC/${PORT}/r0p1"
+            NO_DEFAULT_PATH
+        )
+
+        if(NOT FreeRTOS_${PORT}_PATH)
+            message(WARNING "FreeRTOS port path not found, build might fail")
+        endif()
+
+        list(APPEND FreeRTOS_INCLUDE_DIRS "${FreeRTOS_${PORT}_PATH}")
+        
+        find_file(FreeRTOS_${PORT}_SOURCE
+            NAMES port.c
+            PATHS "${FreeRTOS_${PORT}_PATH}"
+            NO_DEFAULT_PATH
+        )
+        if(NOT (TARGET ${FreeRTOS_NAMESPACE}::${PORT}))
+            add_library(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE IMPORTED)
+            target_link_libraries(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE FreeRTOS)
+            target_sources(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE "${FreeRTOS_${PORT}_SOURCE}")
+            target_include_directories(${FreeRTOS_NAMESPACE}::${PORT} INTERFACE "${FreeRTOS_${PORT}_PATH}")
+        endif()
+        
+        if(FreeRTOS_${PORT}_PATH AND 
+           FreeRTOS_${PORT}_SOURCE AND 
+           FreeRTOS_COMMON_INCLUDE AND
+           FreeRTOS_SOURCE_DIR)
+           set(FreeRTOS_${PORT}_FOUND TRUE)
+        else()
+           set(FreeRTOS_${PORT}_FOUND FALSE)
+        endif()
+    endforeach()
+endmacro()
+
+message(STATUS "Search for FreeRTOS ports: ${FreeRTOS_FIND_COMPONENTS_PORTS}")
+
+if(NOT FreeRTOS_FIND_COMPONENTS_FAMILIES)
+    if(NOT FREERTOS_PATH)
+        set(FREERTOS_PATH /opt/FreeRTOS CACHE PATH "Path to FreeRTOS")
+        message(STATUS "No FREERTOS_PATH specified, using default: ${FREERTOS_PATH}")
```

Comment:
`FREERTOS_PATH` is managed at line 14. I suppose it should not be re-handled here. Please fix.

Reply:
I removed the duplicate handling there